File: algorithm/Python3_version/leetcode1243.py
class Solution:
    def transformArray(self, arr: List[int]) -> List[int]:
        temp = arr.copy()
        while True:
            for i in range(1, len(arr)-1):
                if arr[i-1] > arr[i] and arr[i] < arr[i+1]:
                    temp[i] += 1
                elif arr[i-1] < arr[i] and arr[i] > arr[i+1]:
                    temp[i] -= 1
            if temp == arr:
                break
            arr = temp.copy()
        return arr

File: algorithm/leetcode1243.cpp
class Solution {
public:
    vector<int> transformArray(vector<int>& arr) {
    	vector<int> temp(arr);
    	int i, n = arr.size();
    	while(1)
    	{
    		for(i = 1; i < n-1; ++i)
    		{
    			if(arr[i-1] > arr[i] && arr[i] < arr[i+1])
    				temp[i]++;
    			else if(arr[i-1] < arr[i] && arr[i] > arr[i+1])
    				temp[i]--;
    		}
    		if(temp == arr)
    			break;
    		arr = temp;
    	}
    	return arr;
    }
};